pinot-core/src/main/java/org/apache/pinot/core/operator/InstanceResponseOperator.java

```diff
@@ -111,6 +116,10 @@ private BaseResultsBlock getCombinedResults() {
     try {
       prefetchAll();
       return _combineOperator.nextBlock();
+    } catch (EarlyTerminationException e) {
+      Exception killedErrorMsg = Tracing.getThreadAccountant().getErrorStatus();
+      return new ExceptionResultsBlock(new QueryCancelledException(
+          "Cancelled while combining results" + (killedErrorMsg == null ? StringUtils.EMPTY : " " + killedErrorMsg)));
```

Comment:
So one difference between this and the existing code is that this is not including the original exception which could have been `InterruptedException` or `EarlyTerminationExcetption`. 

The original code was including that inside `QueryCancelledException`, whereas here we have folded both into `EarlyTerminationExcetption`

Reply:
```
`throw new QueryCancelledException(
          "Cancelled while merging results blocks" + (killedErrorMsg == null ? StringUtils.EMPTY
              : " " + killedErrorMsg), e);`
```

Reply:
@Jackie-Jiang IIUC in the runner thread, `InstanceResponseOperator` basically calls `_combineOperator.nextBlock()`; where the `InterruptedException` is folded into `EarlyTerminationException`, right?
It should be fine this way @siddharthteotia as long as `Tracing.getThreadAccountant().getErrorStatus();` get called eventually. It would be able to collect the error status from the accountant.

Reply:
In the original code, we expect all the exceptions being handled within the combine operator, which might not always be the case because interrupt might be handled in `nextBlock()` call instead of the `getNextBlock()`, which will throw `EarlyTerminationException`. In order to handle all of them, we should throw `EarlyTerminationException` out and let the `InstanceResponseOperator` to handle it

Reply:
this is highly unlikely but i was able to fiddle around the underlying threading mechanism to reproduce it :-D.